pubpub/platform
# refactor: remove prisma from stages and pubs pages and make components use suspense more to improve responsiveness

## Issue(s) Resolved

Partially [#586](https://github.com/pubpub/platform/issues/586)

## High-level Explanation of PR
This PR changes the way data is fetched on the `/pubs` and `/stages` pages (not the `/pubs/[pubId]` page mind you).

In general data is now fetched further down the component tree, and split up in more stages in order to leverage `React.Suspense` to show the page to users more quickly.

Before, for e.g. the `stages` page, we did one giant fetch at the top of the page, and drilled that down to components further down the tree by passing them as props.

This had two downsides:
1. The main one being that this fetch takes a very long time, and thus users have to look at nothing or the previous page for quite a long time.
2. Secondly, it makes it slightly harder to reuse components in other places, as they expect the same (part of the) giant payload that is fetched at the start of the `/pubs` and `/stages` page.



### Other changes

#### Move popup

I made the `move` popup slightly nicer. I know we are going to change it soon, which is why i didn't explicitly ask @ktyjscott (although please provide any comments if you have them!).

<img width="567" alt="image" src="https://github.com/user-attachments/assets/06797f66-ebba-4b7e-84ab-6ce15997fc57">

I also added moving indicators

https://github.com/user-attachments/assets/b9bf038b-eb7d-447d-a28b-5b687943d799


#### Run Action in stages
You can now run actions on the `/stages` and `/pubs` pages. This currently adds a TON of extra fetching (for email action, user is fetched for every pub in the stage the instance is in!!!), but I have a PR coming up very soon which addresses that.

## Test Plan

To more clearly show the benefit of this approach, I've created a `kysely` plugin that emulates slow data fetches. 
https://github.com/pubpub/platform/blob/334552a3b7fd57fa65540030c7d12a4c9eefc95d/core/kysely/artificial-latency-plugin.ts#L1-L31

https://github.com/pubpub/platform/blob/334552a3b7fd57fa65540030c7d12a4c9eefc95d/core/kysely/database.ts#L38-L44

It simply adds `env.KYSELY_ARTIFICIAL_LATENCY` milliseconds to the end of every fetch in dev mode, if you have that variable set.

To clearly see the new structure in action, put `KYSELY_ARTIFICIAL_LATENCY=1000` in `.env.local`, then do the following.

1. Go to `/stages`.
2. See first the stages load in
3. Then see the pubs in those stages load in.
4. Do same for `/pubs`

## Screenshots (if applicable)

Loading behavior on `/stages` and `/pubs`



https://github.com/user-attachments/assets/b9a37b42-a099-400a-a9e2-a2653bfa4591





## Notes
